### PR TITLE
Add acceptance tests for moving between integrations.

### PR DESCRIPTION
**Pull request by faun**
**Is Covered by Automated Tests?: YES**

### CODE REVIEW

File: app/components/organizations/integrations/integration-item.js [hunk 1]
```diff
@@ -1,47 +1,11 @@
 import Component from '@ember/component';
 import {lookup} from 'percy-web/lib/computed/objectLookup';
 import {computed} from '@ember/object';
+import AdminMode from 'percy-web/lib/admin-mode';
```
Comment (maprules1000) [inline app/components/organizations/integrations/integration-item.js]: Love it, all about it.

File: app/lib/integration-types.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  GITHUB_ENTERPRISE_INTEGRATION_TYPE,
+  GITLAB_INTEGRATION_TYPE,
+  GITLAB_SELF_HOSTED_INTEGRATION_TYPE,
+} from 'percy-web/models/version-control-integration';
+
+export const INTEGRATION_TYPES = {
+  [GITHUB_INTEGRATION_TYPE]: {
+    textName: 'GitHub',
+    isBeta: false,
+    isGeneralAvailability: true,
```
Comment (maprules1000) [inline app/lib/integration-types.js]: So does this mean there's a state in between beta and generally available?

File: app/templates/organizations/organization/integrations/index.hbs [hunk 3]
```diff
@@ -26,31 +25,14 @@
...
-          organization=model
-        }}
-      {{/if}}
-      {{#if (not model.isGitlabSelfHostedIntegrated)}}
+    <div data-test-integrations-available class="bg-white rounded shadow-1 mb-2">
+      {{#each-in integrationItems as |integrationName integration| }}
         {{organizations/integrations/integration-item
-          integrationName=gitlabSelfHostedIntegrationName
+          integrationName=integrationName
+          isInstalledSection=false
```
Comment (maprules1000) [inline app/templates/organizations/organization/integrations/index.hbs]: I'm not a huge fan of requiring components to know where they sit on the page.

File: app/templates/components/organizations/integrations/integration-item.hbs [hunk 4]
```diff
@@ -1,15 +1,16 @@
-<div class="integration-item flex justify-between items-center p-2 border-b" data-test-integration-item={{routeSlug}}>
+{{#if (or isInstalledSection (not isInstalled))}}
```
Comment (maprules1000) [inline app/templates/components/organizations/integrations/integration-item.hbs]: This is a little confusing. Maybe add a helper to the component and add a comment here describing what's up.

Comment (faun) [reply app/templates/components/organizations/integrations/integration-item.hbs]: See 6391b8d057177d29c6368bada464bb3aebc5df7d

File: app/templates/organizations/organization/integrations/index.hbs (hunk 3 above)
Comment (faun) [reply app/templates/organizations/organization/integrations/index.hbs]: Fixed in 6391b8d057177d29c6368bada464bb3aebc5df7d

File: app/lib/integration-types.js (hunk 2 above)
Comment (faun) [reply app/lib/integration-types.js]: Nope, the isBeta controls if there's a beta badge, and isGeneralAvailability control wether or not you can install it on your own. They're not mutually exclusive. For example, GitHub Enterprise is not in beta, but you can't install it on your own.

**Comment by faun** [comment]
Back at you, @maprules1000

File: app/components/organizations/integrations/integration-item.js [hunk 5]
```diff
@@ -67,6 +32,20 @@ export default Component.extend({
     'organizationModelAttribute',
   ),
 
+  integrationItems: INTEGRATIONS_LOOKUP,
+
+  isEnabled: computed('integrationName', function() {
+    let isGeneralAvailability = this.get('isGeneralAvailability');
+    let isAdminModeEnabled = AdminMode.isAdmin();
+    if (isAdminModeEnabled) {
```
Comment (maprules1000) [inline app/components/organizations/integrations/integration-item.js]: you could simplify this as `isAdminModeEnabled || isGeneralAvailability`

File: app/models/organization.js [hunk 6]
```diff
@@ -60,6 +61,17 @@ export default DS.Model.extend({
   isGitlabSelfHostedIntegrated: bool('gitlabSelfHostedIntegration'),
   isVersionControlIntegrated: bool('versionControlIntegrations.length'),
 
+  availableIntegrations: computed('versionControlIntegrations.[]', function() {
+    let integrations = [];
+    for (const key of Object.keys(INTEGRATION_TYPES)) {
+      let item = INTEGRATION_TYPES[key];
+      if (this.get(`${item.organizationModelAttribute}`) != true) {
+        integrations.push(key);
```
Comment (maprules1000) [inline app/models/organization.js]: 👍

File: app/components/organizations/integrations/integration-item.js (hunk 5 above)
Comment (faun) [reply app/components/organizations/integrations/integration-item.js]: Fixed in 7de5ec7

**Pull request merged**